checker/src/block_visitor.rs

```diff
@@ -267,6 +268,72 @@ impl<'block, 'analysis, 'compilation, 'tcx, E>
         assume!(targets.len() == values.len() + 1);
         let mut default_exit_condition = self.bv.current_environment.entry_condition.clone();
         let discr = self.visit_operand(discr);
+
+        // Check if the discriminant is attached with the ConstantTimeTaint tag.
+        // If so, report an error concerning constant-time security.
+        if self.bv.check_for_errors && self.bv.check_for_constant_time_security {
+            let tag_absence_value =
+                AbstractValue::make_tag_check(discr.clone(), Tag::ConstantTime, false);
+            let (tag_absence_as_bool, entry_cond_as_bool) =
+                self.check_condition_value_and_reachability(&tag_absence_value);
+
+            // We perform the tag check if the current block is possibly reachable.
+            if entry_cond_as_bool.unwrap_or(true) {
+                match tag_absence_as_bool {
+                    None => {
+                        // We cannot decide the result of the tag check.
+                        // In this case, report a warning if the analyzed function is public.
+                        if self.bv.function_being_analyzed_is_root()
+                            || self.bv.preconditions.len() >= k_limits::MAX_INFERRED_PRECONDITIONS
+                        {
```

Comment:
You may want to warn in the case where the condition cannot be promoted because it contains a local variable.